base/applications/kbswitch/indicdll/indicdll.c

```diff
@@ -18,18 +21,18 @@ typedef struct tagSHARED_DATA
     HWND hKbSwitchWnd;
     UINT nHotID;
     DWORD_PTR dwHotMenuItemData;
-    CRITICAL_SECTION csLock;
 } SHARED_DATA, *PSHARED_DATA;
 
 HINSTANCE g_hInstance = NULL;
 HANDLE g_hShared = NULL;
 PSHARED_DATA g_pShared = NULL;
-BOOL g_bCriticalSectionInitialized = 0;
+CRITICAL_SECTION g_csLock;
```

Comment:
To complete what I was saying above, if this whole indicdll code can be run on multiple threads from multiple processes, and the shared-data really has to be shared between all these, then it naively looks to me like acquiring this critical section would just lock within the "current" process using the data, but if some other process is accessing the indicdll stuff, this lock won't be of much effect.

Am I correct, or am I missing/misunderstanding something?

Reply:
Yes, we have to use a mutex instead if exclusive control is needed.